ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m

```diff
@@ -26,22 +28,62 @@ - (void)dealloc {
 - (void)checkAccessState:(void (^)(BOOL))completion {
     self.completion = completion;
 
-    self.publishing = YES;
-    self.service.delegate = self;
-    [self.service publish];
+    BOOL permissionRequestedBefore = [[NSUserDefaults standardUserDefaults] boolForKey:kLocalNetworkPermissionRequestedKey];
 
-    self.timer = [NSTimer scheduledTimerWithTimeInterval:2 repeats:NO block:^(NSTimer * _Nonnull timer) {
-        [self.timer invalidate];
-        self.completion(NO);
-    }];
+    if (!permissionRequestedBefore) {
+        [self logPermissionStatus];
+        [[NSUserDefaults standardUserDefaults] setBool:YES forKey:kLocalNetworkPermissionRequestedKey];
+        [[NSUserDefaults standardUserDefaults] synchronize];
```

Comment:
`synchronize` is no longer necessary to use: https://developer.apple.com/documentation/macos-release-notes/foundation-release-notes?language=objc

Reply:
removed